File: dev-docs/snippets/integrations/call.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Call a third-party integration through the server proxy. `call` is generic,
// so the response `body` is typed `T` and `query` accepts `Record<string, any>`.
export async function call(client: JsBaoClient, integrationKey: string) {
  // #region example
  const response = await client.integrations.call<{ id: string }>({
    integrationKey,
    method: "GET",
    path: "/users/me",
    query: { include: "profile" },
  });
  console.log(response.status, response.body.id);
  // #endregion example
  return response;
}

File: dev-docs/snippets/integrations/call.swift
import JsBaoClient

// Call a third-party integration through the server proxy. The response `body`
// is a `JSONValue` (inspect via accessors/subscripts) and `query` accepts
// `[String: JSONValue]`, mirroring JS's `Record<string, any>`.
func call(client: JsBaoClient, integrationKey: String) async throws {
  // #region example
  let response = try await client.integrations.call(
    IntegrationCallRequest(
      integrationKey: integrationKey,
      method: "GET",
      path: "/users/me",
      query: ["include": "profile"]
    )
  )
  let id = response.body?["id"]?.stringValue
  print(response.status, id ?? "")
  // #endregion example
  _ = response
}
